projector-agent/build.gradle.kts

```diff
@@ -0,0 +1,145 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation. JetBrains designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact JetBrains, Na Hrebenech II 1718/10, Prague, 14000, Czech Republic
+ * if you need additional information or have any questions.
+ */
+
+import java.util.*
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+  kotlin("jvm")
+  `maven-publish`
+}
+
+publishing {
+  publications {
+    create<MavenPublication>("maven") {
+      from(components["java"])
+    }
+  }
+}
+
+kotlin {
+  explicitApi()
+}
+
+val compileKotlin: KotlinCompile by tasks
+val targetJvm: String by project
+compileKotlin.kotlinOptions.jvmTarget = targetJvm
+
+val projectorClientGroup: String by project
+val projectorClientVersion: String by project
+val javassistVersion: String by project
+val agentClassName = "org.jetbrains.projector.agent.MainAgent"
+val launcherClassName = "org.jetbrains.projector.agent.AgentLauncher"
+version = project(":projector-plugin").version
+
+dependencies {
+  implementation("$projectorClientGroup:projector-common:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-server-core:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-util-logging:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-util-loading:$projectorClientVersion")
+  api(project(":projector-awt"))
+  api(project(":projector-server"))
+  implementation("org.javassist:javassist:$javassistVersion")
+}
+
+var serverTargetClasspath: String? = null
+var serverClassToLaunch: String? = null
+var ideaPath: String? = null
+
+rootProject.file("local.properties").let {
+  if (it.canRead()) {
+    ideaPath = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.ideaPath") ?: null
+    serverTargetClasspath = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.targetClassPath") ?: null
+    serverClassToLaunch = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.classToLaunch") ?: null
+  }
+}
```

Comment:
I suggest making the logic nearer to what we had in the past: https://github.com/JetBrains/projector-server/blob/e4f442b130ed46ffd33601e1f068518ad200e6c9/projector-agent/build.gradle#L38-L44

1. Define variables as `val`s to avoid unintended modifications.
2. `?: null` looks redundant (if null then null).
3. Let's load local properties only once as a file-level variable and then get properties from this val.
4. Move `ideaPath` near "Idea launch config".

Another minor improvement: for some reason `jar` is between vars (`serverTargetClasspath` and `serverClassToLaunch`) and their usage ("Agent launch config"), so I suggest rearrange code a bit: move them nearer.